Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -892,7 +900,7 @@ extension BrowserViewController: HomePanelViewControllerDelegate {
         finishEditingAndSubmit(url, visitType: visitType)
     }
     func homePanelViewController(homePanelViewController: HomePanelViewController, didSelectPanel panel: Int) {
-        tabManager.selectedTab?.lastHomePanel = panel
+        tabManager.selectedTab?.webView?.evaluateJavaScript("history.replaceState({}, '', '/about/home/#panel=\(panel)')", completionHandler: nil)
```

Comment:
Note that we show the home panel when we're on about:home, but also when we're on any other page and the user has tapped the URL bar. In the latter case, we don't want to issue a `replaceState` since we're on a regular web page, and this will just end up with a DOM security exception (due to the the cross origin URL). We should guard this line by first checking whether the tab's URL is currently on about:home already; if not, do nothing.